### PR TITLE
AST-394: Add more warnings, errors and info

### PR DESCRIPTION
**Pull request by pierallard**

### CODE REVIEW

File: bin/migrate.php [hunk 1]
```diff
@@ -27,27 +29,54 @@
...
-    new InputArgument('asset-family-code', InputArgument::REQUIRED),
-    new InputArgument('pim-path', InputArgument::REQUIRED)
+    new InputArgument('asset_family_code', InputArgument::REQUIRED),
+    new InputArgument('pim_path', InputArgument::REQUIRED),
+    new InputOption('--help', '-h', InputOption::VALUE_NONE, 'Display this help message'),
 ]);
 
+if (count(array_intersect($_SERVER['argv'], ['-h', '--help']))) {
+    $io->writeln(sprintf("<comment>Description:</comment>
+  Migrate an asset family in full-automatic scenario
```
Comment (SamirBoulil) [inline bin/migrate.php:39]: Link to doc would be cool!

File: bin/migrate.php [hunk 2]
```diff
@@ -27,27 +29,54 @@
...
+
+    die(0);
+};
+
 try {
     $input = new ArgvInput(null, $inputDefinition);
     $input->validate();
-} catch (\Exception $e) {
-    $io->error(sprintf("Input format error: please use %s %s", 'migrate.php', $inputDefinition->getSynopsis(true)));
+} catch (RuntimeException $e) {
```
Comment (SamirBoulil) [inline bin/migrate.php:59]: Apparently we can now catch "Throwable" and it catches EVERY THING even PHP errors

File: bin/migrate.php [hunk 3]
```diff
@@ -27,27 +29,54 @@
...
+$assetFamilyCode = $input->getArgument('asset_family_code');
+$pimPath = $input->getArgument('pim_path');
 
 require dirname(__DIR__).'/config/bootstrap.php';
 
 function executeCommand($arguments, $path, $callback)
 {
     $io = new SymfonyStyle(new ArgvInput(), new ConsoleOutput());
+    if (!empty($path) && !is_dir($path)) {
+        $io->error(sprintf('The folder "%s" is not readable.', $path));
```
Comment (SamirBoulil) [inline bin/migrate.php]:
```suggestion
        $io->error(sprintf('The pim folder "%s" is not accessible.', $path));
```

File: src/Command/CreateFamilyCommand.php [hunk 4]
```diff
@@ -153,58 +154,65 @@ public function execute(InputInterface $input, OutputInterface $output)
...
         $this->io->newLine();
 
-        $this->client->getAssetFamilyApi()->upsert($this->assetFamilyCode, [
-            'code' => $this->assetFamilyCode,
-            'labels' => new \stdClass(), // TODO AST-239 Need to set at least 1 label, else the UI fail :/
-        ]);
+        try {
+            $this->client->getAssetFamilyApi()->upsert($this->assetFamilyCode, [
+                'code' => $this->assetFamilyCode,
+                'labels' => new \stdClass(), // TODO AST-239 Need to set at least 1 label, else the UI fail :/
```
Comment (SamirBoulil) [inline src/Command/CreateFamilyCommand.php]: We can remove the TODO here

File: src/Command/MigrateCommand.php [hunk 5]
```diff
@@ -505,13 +503,13 @@ private function migrate(
             } else if ($referenceType === self::AUTO) {
                 $referenceType = $this->guessReferenceType($assetCsvFilename);
             }
-            $this->io->writeln(sprintf('The command will be ran with <options=bold>--reference-type=%s</>', $referenceType));
+            $this->io->writeln(sprintf('The command will be run with <options=bold>--reference_type=%s</>', $referenceType));
```
Comment (SamirBoulil) [inline src/Command/MigrateCommand.php:506]: Suggestion
```suggestion
            $this->io->writeln(sprintf('The command will run with <options=bold>--reference_type=%s</>', $referenceType));
```

**Review by SamirBoulil — APPROVED** [review]

File: bin/migrate.php (hunk 1 above)
Comment (pierallard) [reply bin/migrate.php:39]: There are already 4 links in the README, isn't it OK?

File: bin/migrate.php (hunk 2 above)
Comment (pierallard) [reply bin/migrate.php:59]: I don't want to catch EVERYTHING, I just want to catch the \Exception raised by the ->validate

File: src/Command/MigrateCommand.php (hunk 5 above)
Comment (pierallard) [reply src/Command/MigrateCommand.php:506]: Camille has already reviewed the english

File: bin/migrate.php (hunk 1 above)
Comment (SamirBoulil) [reply bin/migrate.php:39]: NO ITS NOT OK NOT GTM PIERRE YOUR WORK IS SO NOT LINKABLE

Comment (SamirBoulil) [reply bin/migrate.php:39]: (ok no link)

File: bin/migrate.php (hunk 2 above)
Comment (SamirBoulil) [reply bin/migrate.php:59]: ouki

File: src/Command/MigrateCommand.php (hunk 5 above)
Comment (SamirBoulil) [reply src/Command/MigrateCommand.php:506]: ouki

**Review by juliensnz — APPROVED** [review]

**Pull request merged**